src/app/zap-templates/zcl/data-model/chip/matter-devices.xml

```diff
@@ -2209,6 +2209,12 @@ limitations under the License.
         <deviceId editable="false">0x0070</deviceId>
         <class>Simple</class>
         <scope>Endpoint</scope>
+        <endpointComposition>
+            <compositionType>tree</compositionType>
+            <endpoint mandatory="true" constraint="1">
```

Comment:
What does `constraint="1"` mean here?

Reply:
And don't we need to at least define the "full" composition bits in the XML?

Reply:
> What does `constraint="1"` mean here?

The idea is that you need 1
I understand things could get more complicated. Perhaps I change it to "min 1" and let ZAP consume exactly the value that is in the spec?

Reply:
> And don't we need to at least define the "full" composition bits in the XML?

ZAP consumes whatever is set as <compositionType>. It did not occur to me that it needed to be defined. I guess it would make sense to do <compositionType fullFamily = false tree = true> as an alternative. Either way is fine with me.

Reply:
@bzbarsky-apple I made some changes to address your concerns